src/rule.ts

```diff
@@ -344,7 +344,7 @@ function executeRule(rule: Rule, obj: Entity): void {
 	});
 };
 
-export function Rule$ensureConditionType<DesiredConditionType = ErrorConditionType | WarningConditionType>(ruleName: string, typeOrProp: Type | Property, category: string = "Error"): ErrorConditionType | WarningConditionType {
+export function Rule$ensureConditionType<DesiredConditionType = ErrorConditionType | WarningConditionType>(rule: Rule, ruleName: string, typeOrProp: Type | Property, category: string = "Error"): ErrorConditionType | WarningConditionType {
```

Comment:
One final pre-existing issue, the `<DesiredConditionType = ErrorConditionType | WarningConditionType>` generic type parameter doesn't make sense and is not needed.

Reply:
fixed!